sklearn/linear_model/_sgd_fast.pyx

```diff
@@ -680,7 +680,13 @@ def _plain_sgd(cnp.ndarray[double, ndim=1, mode='c'] weights,
 
     w.reset_wscale()
 
-    return weights, intercept, average_weights, average_intercept, epoch + 1
+    return (
+        np.asarray(weights),
+        intercept,
+        np.asarray(average_weights),
+        average_intercept,
+        epoch + 1
+    )
```

Comment:
`np.asarray` not only creates interactions with the CPython but it also might create a copy.

We better return the return objects, that is the original NumPy arrays, using the `base` attribute of memory-views.
```suggestion
    return (
        weights.base,
        intercept,
        average_weights.base,
        average_intercept,
        epoch + 1
    )
```